lib/reverse_sentence.rb

```diff
@@ -1,6 +1,44 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is num of elements in string
+# Space complexity: O(1) no new space is created in algorithm
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  # nil check 
+  return my_sentence if my_sentence == nil || my_sentence.length == 0
+
+  # counter for word reverse; space before first character
+  left = 0
+  right = my_sentence.length - 1
```

Comment:
I like these variable names! I can imagine left and right clearly moving across the sentence.